### ISSUE TITLE
Can't run tests in netcoreapp1.1 project

### ISSUE THREAD

**Issue by alexvaluyskiy** [warrenfalk/rocksdb-sharp#19]
I've created a test project and have added `RocksDbNative` package there. But I can't run tests. I've tried many different options
- Resharper Test Runner
- VS 2017 test runner
- dotnet xunit
- dotnet test

Exception was thrown
```
    TestExample.TestRocks.Test [FAIL]
      System.TypeInitializationException : The type initializer for 'RocksDbSharp.Native' threw an exception.
      ---- NativeImport.NativeLoadException : Unable to locate rocksdb native library, either install it, or use RocksDbNative nuget package
Searched:C:\Users\ravenger\.nuget\packages\dotnet-xunit\2.3.0-beta2-build3683\tools\netcoreapp1.0\native\amd64\rocksdb-5.4.6.dll: (Win32Exception) The specified module could not be found
C:\Users\ravenger\.nuget\packages\dotnet-xunit\2.3.0-beta2-build3683\tools\netcoreapp1.0\native\amd64\rocksdb-5.4.dll: (Win32Exception) The specified module could not be found
C:\Users\ravenger\.nuget\packages\dotnet-xunit\2.3.0-beta2-build3683\tools\netcoreapp1.0\native\amd64\rocksdb-5.dll: (Win32Exception) The specified module could not be found
C:\Users\ravenger\.nuget\packages\dotnet-xunit\2.3.0-beta2-build3683\tools\netcoreapp1.0\native\amd64\rocksdb.dll: (Win32Exception) The specified module could not be found
C:\Users\ravenger\.nuget\packages\dotnet-xunit\2.3.0-beta2-build3683\tools\netcoreapp1.0\native\rocksdb-5.4.6.dll: (Win32Exception) The specified module could not be found
C:\Users\ravenger\.nuget\packages\dotnet-xunit\2.3.0-beta2-build3683\tools\netcoreapp1.0\native\rocksdb-5.4.dll: (Win32Exception) The specified module could not be found
C:\Users\ravenger\.nuget\packages\dotnet-xunit\2.3.0-beta2-build3683\tools\netcoreapp1.0\native\rocksdb-5.dll: (Win32Exception) The specified module could not be found
C:\Users\ravenger\.nuget\packages\dotnet-xunit\2.3.0-beta2-build3683\tools\netcoreapp1.0\native\rocksdb.dll: (Win32Exception) The specified module could not be found
C:\Users\ravenger\.nuget\packages\dotnet-xunit\2.3.0-beta2-build3683\tools\netcoreapp1.0\amd64\rocksdb-5.4.6.dll: (Win32Exception) The specified module could not be found
C:\Users\ravenger\.nuget\packages\dotnet-xunit\2.3.0-beta2-build3683\tools\netcoreapp1.0\amd64\rocksdb-5.4.dll: (Win32Exception) The specified module could not be found
C:\Users\ravenger\.nuget\packages\dotnet-xunit\2.3.0-beta2-build3683\tools\netcoreapp1.0\amd64\rocksdb-5.dll: (Win32Exception) The specified module could not be found
C:\Users\ravenger\.nuget\packages\dotnet-xunit\2.3.0-beta2-build3683\tools\netcoreapp1.0\amd64\rocksdb.dll: (Win32Exception) The specified module could not be found
C:\Users\ravenger\.nuget\packages\dotnet-xunit\2.3.0-beta2-build3683\tools\netcoreapp1.0\rocksdb-5.4.6.dll: (Win32Exception) The specified module could not be found
C:\Users\ravenger\.nuget\packages\dotnet-xunit\2.3.0-beta2-build3683\tools\netcoreapp1.0\rocksdb-5.4.dll: (Win32Exception) The specified module could not be found
C:\Users\ravenger\.nuget\packages\dotnet-xunit\2.3.0-beta2-build3683\tools\netcoreapp1.0\rocksdb-5.dll: (Win32Exception) The specified module could not be found
C:\Users\ravenger\.nuget\packages\dotnet-xunit\2.3.0-beta2-build3683\tools\netcoreapp1.0\rocksdb.dll: (Win32Exception) The specified module could not be found
rocksdb-5.4.6.dll: (Win32Exception) The specified module could not be found
rocksdb-5.4.dll: (Win32Exception) The specified module could not be found
rocksdb-5.dll: (Win32Exception) The specified module could not be found
rocksdb.dll: (Win32Exception) The specified module could not be found
C:\Users\ravenger\.nuget\packages\rocksdbsharp\5.4.6\lib\netstandard1.6\native\amd64\rocksdb-5.4.6.dll: (Win32Exception) The specified module could not be found
C:\Users\ravenger\.nuget\packages\rocksdbsharp\5.4.6\lib\netstandard1.6\native\amd64\rocksdb-5.4.dll: (Win32Exception) The specified module could not be found
C:\Users\ravenger\.nuget\packages\rocksdbsharp\5.4.6\lib\netstandard1.6\native\amd64\rocksdb-5.dll: (Win32Exception) The specified module could not be found
C:\Users\ravenger\.nuget\packages\rocksdbsharp\5.4.6\lib\netstandard1.6\native\amd64\rocksdb.dll: (Win32Exception) The specified module could not be found
C:\Users\ravenger\.nuget\packages\rocksdbsharp\5.4.6\lib\netstandard1.6\native\rocksdb-5.4.6.dll: (Win32Exception) The specified module could not be found
C:\Users\ravenger\.nuget\packages\rocksdbsharp\5.4.6\lib\netstandard1.6\native\rocksdb-5.4.dll: (Win32Exception) The specified module could not be found
C:\Users\ravenger\.nuget\packages\rocksdbsharp\5.4.6\lib\netstandard1.6\native\rocksdb-5.dll: (Win32Exception) The specified module could not be found
C:\Users\ravenger\.nuget\packages\rocksdbsharp\5.4.6\lib\netstandard1.6\native\rocksdb.dll: (Win32Exception) The specified module could not be found
C:\Users\ravenger\.nuget\packages\rocksdbsharp\5.4.6\lib\netstandard1.6\amd64\rocksdb-5.4.6.dll: (Win32Exception) The specified module could not be found
C:\Users\ravenger\.nuget\packages\rocksdbsharp\5.4.6\lib\netstandard1.6\amd64\rocksdb-5.4.dll: (Win32Exception) The specified module could not be found
C:\Users\ravenger\.nuget\packages\rocksdbsharp\5.4.6\lib\netstandard1.6\amd64\rocksdb-5.dll: (Win32Exception) The specified module could not be found
C:\Users\ravenger\.nuget\packages\rocksdbsharp\5.4.6\lib\netstandard1.6\amd64\rocksdb.dll: (Win32Exception) The specified module could not be found
C:\Users\ravenger\.nuget\packages\rocksdbsharp\5.4.6\lib\netstandard1.6\rocksdb-5.4.6.dll: (Win32Exception) The specified module could not be found
C:\Users\ravenger\.nuget\packages\rocksdbsharp\5.4.6\lib\netstandard1.6\rocksdb-5.4.dll: (Win32Exception) The specified module could not be found
C:\Users\ravenger\.nuget\packages\rocksdbsharp\5.4.6\lib\netstandard1.6\rocksdb-5.dll: (Win32Exception) The specified module could not be found
C:\Users\ravenger\.nuget\packages\rocksdbsharp\5.4.6\lib\netstandard1.6\rocksdb.dll: (Win32Exception) The specified module could not be found
rocksdb-5.4.6.dll: (Win32Exception) The specified module could not be found
rocksdb-5.4.dll: (Win32Exception) The specified module could not be found
rocksdb-5.dll: (Win32Exception) The specified module could not be found
rocksdb.dll: (Win32Exception) The specified module could not be found
      Stack Trace:
        C:\Users\wfalk\source\rocksdb-sharp\RocksDbSharp\Options.cs(26,0): at RocksDbSharp.OptionsHandle..ctor()
           at RocksDbSharp.DbOptions..ctor()
        C:\Projects\test\RocksDbTestSample\TestRocks.cs(11,0): at TestExample.TestRocks.Test()
        ----- Inner Stack Trace -----
           at NativeImport.Importers.Import[T](INativeLibImporter importer, String libName, String version, Boolean suppressUnload)
        C:\Users\wfalk\source\rocksdb-sharp\RocksDbSharp\AutoNativeImport.cs(39,0): at NativeImport.Auto.Import[T](String name, String version, Boolean suppressUnload)
        C:\Users\wfalk\source\rocksdb-sharp\RocksDbSharp\Native.Load.cs(18,0): at RocksDbSharp.Native..cctor()
  Finished:    RocksDbTestSample
```
I've tried 5.2.1 and 5.4.6 versions
There is my project: https://github.com/alexvaluyskiy/RocksDbTestSample

**Comment by alexvaluyskiy** [comment]
It's using a weird path ` C:\Users\wfalk`. It is not my user :)

**Comment by warrenfalk** [comment]
@alexvaluyskiy, the paths to source files in stack traces reflect the absolute path to the source at build time.  I don't think that is particularly weird.  The exception dump you provided seems to say it only searched inside ```C:\Users\ravenger``` which is what I would expect.

I'll look into the search paths used during unit tests.

**Comment by warrenfalk** [comment]
I think the issue is that the CodeBase for the RocksDbSharp dll is not the codebase that the RocksDbNative contents are necessarily copied to.  This is especially true in unit test environments, which is why this doesn't work.

Fundamentally this appears to be a flaw in the current strategy of the RocksDbNative package as just a content package, so I will modify that.  I will add a very small managed assembly as output of a new RocksDbNative project.  The native binaries will then be seen as content for that assembly and should therefore be deployed to the CodeBase for that assembly.  I can query that assembly for its CodeBase from RocksDbSharp when searching for native binaries.

**Comment by warrenfalk** [comment]
This is resolved in the 5.4.6.7 which is a pre-release.  Once tested better, I'll release it as stable.
